go-controller/pkg/clustermanager/default_network_cluster_controller.go

```diff
@@ -0,0 +1,400 @@
+
+	// retry framework for nodes
+	retryNodes *objretry.RetryFramework
+}
+
+// defaultNetworkClusterController is the cluster controller for the default network
+// for all the nodes. Listens to the node events and allocates subnet from the
+// default cluster subnet pool.
+type defaultNetworkClusterController struct {
+	networkClusterControllerBase
+	defaultNetworkController *NodeNetworkController
+
+	hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+}
+
+func newDefaultNetworkClusterController(ovnClient *util.OVNClusterManagerClientset, wf *factory.WatchFactory) *defaultNetworkClusterController {
+
+	kube := &kube.Kube{
+		KClient: ovnClient.KubeClient,
+	}
+
+	stopChan := make(chan struct{})
+	wg := &sync.WaitGroup{}
+
+	var hybridOverlaySubnetAllocator *subnetallocator.HostSubnetAllocator
+	if config.HybridOverlay.Enabled {
+		hybridOverlaySubnetAllocator = subnetallocator.NewHostSubnetAllocator()
+	}
+	ncm := &defaultNetworkClusterController{
+		networkClusterControllerBase: networkClusterControllerBase{
+			kube:         kube,
+			watchFactory: wf,
+			stopChan:     stopChan,
+			wg:           wg,
+		},
+		defaultNetworkController:     newNodeNetworkController(kube, wf, ovntypes.DefaultNetworkName),
+		hybridOverlaySubnetAllocator: hybridOverlaySubnetAllocator,
+	}
+
+	ncm.initRetryFramework()
+	return ncm
+}
+
+func (ncm *defaultNetworkClusterController) initRetryFramework() {
+	ncm.retryNodes = ncm.newRetryFramework(factory.NodeType)
+}
+
+// Init initializes the default network subnet allocator ranges
+// and hybrid network subnet allocator ranges if hybrid overlay is enabled.
+func (ncm *defaultNetworkClusterController) Init() error {
+	if err := ncm.defaultNetworkController.InitSubnetAllocatorRanges(config.Default.ClusterSubnets); err != nil {
+		return fmt.Errorf("failed to initialize host subnet allocator ranges: %w", err)
+	}
+
+	if config.HybridOverlay.Enabled {
+		if err := ncm.hybridOverlaySubnetAllocator.InitRanges(config.HybridOverlay.ClusterSubnets); err != nil {
+			return fmt.Errorf("failed to initialize hybrid overlay subnet allocator ranges: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (ncm *defaultNetworkClusterController) Stop() {
+	close(ncm.stopChan)
+	ncm.wg.Wait()
+}
+
+// Run starts watching kubernetes nodes.
+func (ncm *defaultNetworkClusterController) Run() error {
+	return ncm.watchNodes()
+}
+
+// WatchNodes starts the watching of node resource and calls
+// back the appropriate handler logic
+func (ncm *defaultNetworkClusterController) watchNodes() error {
+	if ncm.nodeHandler != nil {
+		return nil
+	}
+
+	handler, err := ncm.retryNodes.WatchResource()
+	if err != nil {
+		return err
+	}
+
+	ncm.nodeHandler = handler
+	return nil
+}
+
+// defaultNetworkClusterControllerEventHandler object handles the events
+// from retry framework.
+type defaultNetworkClusterControllerEventHandler struct {
+	objretry.EventHandler
+
+	objType  reflect.Type
+	ncm      *defaultNetworkClusterController
```

Comment:
Do you think this would be cleaner to implement `EventHandler` using type `defaultNetworkClusterController` and just move the fields you require in `defaultNetworkClusterControllerEventHandler` into that type?

Reply:
I  dont like `defaultNetworkClusterController` referenced here also.

Reply:
hmm maybe the more reviews I do I will get used to how its being done everywhere in the code.. 
but if this same pattern is being done everywhere and easier for everyone then ignore my comment, its possible I am the only one drowning with these struct dependencies :D